### PR TITLE
APM-4931: Request and response sources for One Klarna added

### PR DESCRIPTION
**Pull request by michael-zhukov-cko**
# Description of changes in PR

One Klarna Api reference added 

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [ ] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

File: nas_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                type: string
+                description: 'ISO 3166 alpha-2 account holder country code.'
+            description: 'Address of the account holder.'
+          email:
+            type: string
+            description: 'Email address of the account holder.'
+          date_of_birth:
+            type: string
+          gender:
+            type: string
```
Comment (Enrica-Mameli-cko) [inline nas_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:58]: date_of_birth and gender do not have any description

**Review by Enrica-Mameli-cko — APPROVED** [review]
left some minor comment on descriptions that should be completed before merging

File: nas_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml (hunk 1 above)
Comment (michael-zhukov-cko) [reply nas_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:59]: fixed

**Review by alex-larin-cko — DISMISSED** [review]

**Review by Enrica-Mameli-cko — APPROVED** [review]

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 2]
```diff
@@ -8,4 +8,51 @@ properties:
...
-    example: 'cartes_bancaires'
\ No newline at end of file
+    example: 'cartes_bancaires'
+  tax_amount:
+    type: number
+    description: Total tax amount of the order.
+    example: 1000
+  purchase_country:
+    type: string
+    description: 'The purchase country of the customer. ISO 3166 alpha-2 purchase country'
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:18]:
```suggestion
    description: 'The purchase country of the customer, as an ISO 3166 alpha-2 country code.'
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 3]
```diff
@@ -8,4 +8,51 @@ properties:
...
+  locale:
+    type: string
+    pattern: '^[a-z]{2}(?:-[A-Z][a-z]{3})?(?:-(?:[A-Z]{2}))?$'
+    description: ISO 639-2 language code
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:22]:
```suggestion
    description: The language, as an ISO 639-2 code.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 4]
```diff
@@ -8,4 +8,51 @@ properties:
...
+    example: en-US
+    maxLength: 10
+    minLength: 2
+  partner_order_id:
+    type: string
+    description: The Klarna Order Id associated for the payment
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:28]:
```suggestion
    description: The Klarna order ID associated with the payment.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 5]
```diff
@@ -8,4 +8,51 @@ properties:
...
+  fraud_status:
+    type: string
+    description: The Klarna Fraud Status associated for the payment
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:31]:
```suggestion
    description: The Klarna fraud status associated with the payment.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 6]
```diff
@@ -8,4 +8,51 @@ properties:
...
+  provider_authorized_payment_method:
+    x-klarna-name: initial_payment_method
+    type: object
+    description: Initial/Authorized payment method.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:35]:
```suggestion
    description: The initial or authorized payment method.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 7]
```diff
@@ -8,4 +8,51 @@ properties:
...
+    required:
+      - type
+    properties:
+      type:
+        type: string
+        description: Payment option type name
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:41]:
```suggestion
        description: The payment option's type name.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 8]
```diff
@@ -8,4 +8,51 @@ properties:
...
+      description:
+        type: string
+        description: Payment option description
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:44]:
```suggestion
        description: The payment option's description.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 9]
```diff
@@ -8,4 +8,51 @@ properties:
...
+      number_of_installments:
+        type: integer
+        description: Payment option number of installments
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:47]:
```suggestion
        description: The payment option's number of installments.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 10]
```diff
@@ -8,4 +8,51 @@ properties:
...
+      number_of_days:
+        type: integer
+        description: Payment option number of installments
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:50]:
```suggestion
        description: The payment option's number of days.
```

Comment (sebastian-garcia-cardona-cko) [reply nas_spec/components/schemas/Payments/ProcessingData.yaml:50]: Worth adding to this description i.e. is it number of days to pay or something else?

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 11]
```diff
@@ -8,4 +8,51 @@ properties:
...
+  custom_payment_method_ids:
+    type: array
+    description: >-
+      The array could be used to define which of the configured payment
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:54]:
```suggestion
      An array defining which of the configured payment options within a payment category (for example, `pay_later` or `pay_over_time`) should be displayed for this purchase.
```

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 12]
```diff
@@ -8,4 +8,51 @@ properties:
...
+      options within a payment category (pay_later, pay_over_time, etc.)
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:55]: Delete if above change is accepted.

File: nas_spec/components/schemas/Payments/ProcessingData.yaml [hunk 13]
```diff
@@ -8,4 +8,51 @@ properties:
...
+      should be shown for this purchase.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingData.yaml:56]: As above, delete if single-line description is accepted.

File: nas_spec/components/schemas/Payments/ProcessingSettings.yaml [hunk 14]
```diff
@@ -267,3 +267,29 @@ properties:
...
                 type: string
+  purchase_country:
+    type: string
+    description: ISO 3166 alpha-2 purchase country code.
+    example: GB
+  custom_payment_method_ids:
+    x-klarna-name: custom_payment_method_ids
+    x-cko-passthrough: true
+    x-klarna-docs: https://developers.klarna.com/api/#payments-api__create-a-new-credit-session__custom_payment_method_ids
+    description: The array could be used to define which of the configured payment options within a payment category (pay_later, pay_over_time, etc.) should be shown for this purchase.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingSettings.yaml:278]:
```suggestion
    description: An array defining which of the configured payment options within a payment category (for example, `pay_later` or `pay_over_time`) should be displayed for this purchase.
```

File: nas_spec/components/schemas/Payments/ProcessingSettings.yaml [hunk 15]
```diff
@@ -267,3 +267,29 @@ properties:
...
+    type: array
+    items:
+      type: string
+  merchant_callback_url:
+    type: string
+    description: >-
+      URL which be used to notify the client that the order has been created.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingSettings.yaml:285]:
```suggestion
      A URL which you can use to notify the customer that the order has been created.
```

File: nas_spec/components/schemas/Payments/ProcessingSettings.yaml [hunk 16]
```diff
@@ -267,3 +267,29 @@ properties:
...
+  shipping_delay :
+    type: integer
+    description: 'Delay before the order will be shipped.'
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingSettings.yaml:288]:
```suggestion
    description: 'The delay before the order will be shipped.'
```

Comment (sebastian-garcia-cardona-cko) [reply nas_spec/components/schemas/Payments/ProcessingSettings.yaml:288]: We should also add whether this integer represents calendar days, working days, etc.

File: nas_spec/components/schemas/Payments/ProcessingSettings.yaml [hunk 17]
```diff
@@ -267,3 +267,29 @@ properties:
...
+  shipping_info:
+    x-klarna-docs: https://developers.klarna.com/api/#order-management-api__create-capture__shipping_info
+    x-cko-passthrough: true
+    type: array
+    description: Shipping information for this capture. Maximum 500 items.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/ProcessingSettings.yaml:293]:
```suggestion
    description: The shipping information for the payment capture. Maximum 500 items.
```

**Review by sebastian-garcia-cardona-cko — CHANGES_REQUESTED** [review]

File: nas_spec/components/schemas/Payments/ProcessingData.yaml (hunk 10 above)
Comment (alex-larin-cko) [reply nas_spec/components/schemas/Payments/ProcessingData.yaml:50]: @sebastian-garcia-cardona-cko , yes, this is the frequency of the intervals of the payment

**Review by sebastian-garcia-cardona-cko — APPROVED** [review]

**Review by Enrica-Mameli-cko — DISMISSED** [review]

**Review by sebastian-garcia-cardona-cko — APPROVED** [review]

**Review by Enrica-Mameli-cko — APPROVED** [review]

**Pull request merged**